charts/eks/templates/rbac/clusterrole.yaml

```diff
@@ -18,7 +18,9 @@ rules:
     resources: ["nodes", "nodes/status"]
     verbs: ["get", "watch", "list"]
   - apiGroups: [""]
-    resources: ["pods", "nodes/proxy", "nodes/metrics", "nodes/stats"]
+    resources: [ "pods",{{- if .Values.isolation.enabled }}
+      {{- if .Values.isolation.nodeProxyPermission.enabled}} "node/proxy",{{- end}}
+      {{- else}} "node/proxy",{{- end }} "nodes/metrics", "nodes/stats"]
```

Comment:
I know the PR is still in draft state, but adding my 2cents to potentially save one review iteration :) 
I'd rather see `"nodes/proxy"` permission in a separate rule, which can then be wrapped in conditions. That would be more readable.

Reply:
Have added a new rule for node/proxy permission for readability.